.github/workflows/update_llm_perf_cpu_pytorch.yaml

```diff
@@ -4,16 +4,32 @@ on:
   workflow_dispatch:
   schedule:
     - cron: "0 0 * * *"
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - labeled
+      - unlabeled
 
 concurrency:
   cancel-in-progress: true
-  group: ${{ github.workflow }}-${{ github.ref }}
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
 
 env:
   IMAGE: ghcr.io/huggingface/optimum-benchmark:latest-cpu
 
 jobs:
   run_benchmarks:
+    if: ${{
+      (github.event_name == 'push') ||
+      (github.event_name == 'workflow_dispatch') ||
+      contains( github.event.pull_request.labels.*.name, 'leaderboard')}}
```

Comment:
you can probably add more specifications here to be able to run specific benchmarks, like `cuda`/`cpu`
didn't try it, but you might also be able to add conditions on matrix arguments, like `|| contains( github.event.pull_request.labels.*.name, matrix.subset)}}` to run specific subsets or specific machines